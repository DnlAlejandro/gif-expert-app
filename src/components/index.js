export * from './AddCategories';
export * from './GifGrid';
export * from './GifItem';