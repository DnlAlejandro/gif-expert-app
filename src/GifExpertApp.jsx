import { useState } from "react";
import { AddCategories, GifGrid } from "./components";

export const GifExpertApp = () => {
    const [categories, setCategories] = useState(["UFC"]);

    const onAddCategory = (newCategory) => {
        if (categories.includes(newCategory)) return;
        setCategories([newCategory, ...categories]);
    };

    return (
        <>
            <h1>GifExpertApp</h1>
            <AddCategories
                //setCategories={setCategories}
                onNewCategory={onAddCategory}
                currentCategories={categories}
            />
            {categories.map((category) => (
                <GifGrid 
                    key={category} 
                    category={category}
                />
                    // <div key={category}>
                    //     <h3>{category}</h3>
                    //     <li>{category}</li>
                    // </div>
                )
            )}
        </>
    );
};
