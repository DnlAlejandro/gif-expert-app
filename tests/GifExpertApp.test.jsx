import { render, screen } from "@testing-library/react"
import { GifExpertApp } from "../src/GifExpertApp"

describe('test in GifExpertApp', () => { 

    test("should be match with snapshot", () => {
        const { container } = render(<GifExpertApp />);
        expect(container).toMatchSnapshot();
        console.log(container)
    });

    test('should exist title page', () => { 
        render(<GifExpertApp />);
        expect(screen.getByText('GifExpertApp')).toBeTruthy();
    })

    test('should return initial state', () => {
        const a = render(<GifExpertApp />);
    
    });
})