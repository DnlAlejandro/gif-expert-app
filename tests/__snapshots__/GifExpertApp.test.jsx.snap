// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test in GifExpertApp should be match with snapshot 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="search gifs"
      type="text"
      value=""
    />
  </form>
  <h3>
    UFC
  </h3>
  <h2>
    Loading...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
